src/main/scala/regmapper/RegisterCrossing.scala

```diff
@@ -7,129 +7,167 @@ import chisel3.util.{Irrevocable}
 import util.{AsyncQueue,AsyncScope,AsyncResetRegVec}
 
 // A very simple flow control state machine, run in the specified clock domain
-class BusyRegisterCrossing(clock: Clock, reset: Bool)
-    extends Module(_clock = clock, _reset = reset) {
+class BusyRegisterCrossing extends Module {
   val io = new Bundle {
-    val progress       = Bool(INPUT)
-    val request_valid  = Bool(INPUT)
-    val response_ready = Bool(INPUT)
-    val busy           = Bool(OUTPUT)
+    val bypass                  = Bool(INPUT)
+    val master_request_valid    = Bool(INPUT)
+    val master_request_ready    = Bool(OUTPUT)
+    val master_response_valid   = Bool(OUTPUT)
+    val master_response_ready   = Bool(INPUT)
+    val crossing_request_valid  = Bool(OUTPUT)
+    val crossing_request_ready  = Bool(INPUT)
+    // ... no crossing_response_ready; we are always ready
   }
 
-  val busy = RegInit(Bool(false))
-  when (io.progress) {
-    busy := Mux(busy, !io.response_ready, io.request_valid)
+  val busy   = RegInit(Bool(false))
+  val bypass = Reg(Bool())
+
+  when (io.crossing_request_ready || Mux(busy, bypass, io.bypass)) {
+    busy := Mux(busy, !io.master_response_ready, io.master_request_valid)
+  }
+
+  when (io.master_request_valid && io.master_request_ready) {
+    bypass := io.bypass
   }
-  io.busy := busy
+
+  io.crossing_request_valid := io.master_request_valid && !io.bypass && !busy
+  io.master_request_ready   := (io.bypass || io.crossing_request_ready) && !busy
+  io.master_response_valid  := (bypass    || io.crossing_request_ready) &&  busy
+}
+
+class RegisterCrossingAssertion extends Module {
+  val io = new Bundle {
+    val master_bypass = Bool(INPUT)
+    val slave_reset = Bool(INPUT)
+  }
+
+  assert (io.master_bypass || !io.slave_reset)
 }
 
 // RegField should support connecting to one of these
 class RegisterWriteIO[T <: Data](gen: T) extends Bundle {
-  val request  = Decoupled(gen).flip()
+  val request  = Decoupled(gen).flip
   val response = Irrevocable(Bool()) // ignore .bits
 }
 
-// To turn on/off a domain:
-//  1. lower allow on the other side
-//  2. wait for inflight traffic to resolve
-//  3. assert reset in the domain
-//  4. turn off the domain
-//  5. turn on the domain
-//  6. deassert reset in the domain
-//  7. raise allow on the other side
+// To turn off=>on a domain:
+//  A. To turn disable the master domain
+//    1. wait for all inflight traffic to resolve
+//    2. assert master reset
```

Comment:
You don't have to. The instructions were for how to disable a domain, which involves using bypass. You can set bypass and not reset if you choose.